src/index.js

```diff
@@ -57,8 +60,22 @@ export default function rehypePrettyCode(options = {}) {
     getHighlighter = shikiHighlighter,
   } = options;
 
-  // Cache highlighters per unified processor
-  const highlighterCache = new Map();
+  const optionsHash = hashObj(
+    {
+      theme,
+      tokensMap,
+      onVisitLine,
+      onVisitHighlightedLine,
+      onVisitHighlightedWord,
+      getHighlighter,
+    },
+    {algorithm: 'sha1'}
+  );
+  let highlighterCache = themeCache.get(`${optionsHash}.highlighterCache`);
```

Comment:
Can you just do `optionsCache.get(optionsHash)`? Which will return the highlighter cache you want?

Reply:
Yeah, good idea.